DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -80,93 +125,124 @@ function Test-TargetResource
         [String]$InterfaceAlias,
 
         [Parameter(Mandatory)]
-        [ValidateSet("IPv4", "IPv6")]
+        [ValidateSet('IPv4', 'IPv6')]
         [String]$AddressFamily
     )
+    # Flag to signal whether settings are correct
+    [Boolean]$requiresChanges = $false
 
-    Test-Properties @PSBoundParameters
-}
+    try
+    {        
+        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
+            'Checking the DNS Server Address ...' 
+            ) -join '' )
+
+        #Validate the Settings passed
+        Foreach ($ServerAddress in $Address) {       
+            Test-ResourceProperty `
+                -Address $ServerAddress `
+                -AddressFamily $AddressFamily `
+                -InterfaceAlias $InterfaceAlias
+        }
 
+        #Get the current DNS Server Addresses based on the parameters given.
+        $currentAddress = (Get-DnsClientServerAddress `
+            -InterfaceAlias $InterfaceAlias `
+            -AddressFamily $AddressFamily `
+            -ErrorAction Stop).ServerAddresses
+
+        #Check if the Server addresses are the same as the desired addresses.
+        if(@(Compare-Object `
```

Comment:
nit: I found this a little hard to read, could we assign this to a variable and if/else on that?

ie: 

````powershell
        $addressCompare = (Compare-Object -ReferenceObject $currentAddress `
            -DifferenceObject $Address -SyncWindow 0).Length -gt 0

        if ($addressCompare)
````